mac_serial_ng.py

```diff
@@ -0,0 +1,755 @@
+        url = f"https://{server_rib}/redfish/v1/"
+        my_response = requests.get(url, auth=(username, password), verify=False)
+        data = my_response.json()
+        return data["Links"]["Sessions"]["@odata.id"]
+
+
+    def session_create_x_auth_token(self, server_rib: str, username: str, password: str, session_uri: str) -> Tuple[str, Optional[str]]:
+        url = f"https://{server_rib}{session_uri}"
+        payload = {"UserName": username, "Password": password}
+        headers = {'content-type': 'application/json'}
+        response = requests.post(url, json=payload, headers=headers, verify=False)
+        if response.headers.get("x-auth-token") is None:
+            logging.error("Not able to create token")
+            sys.exit(1)
+        data = response.json()
+        my_session_id = data["Id"]
+        my_session_token = response.headers.get("x-auth-token")
+        return my_session_id, my_session_token
+
+
+    def session_delete_x_auth_session(self, server_rib: str, session_x_auth_token: str, session_uri: str, session_id: str):
+        url = f"https://{server_rib}{session_uri}/{session_id}"
+        requests.delete(url, headers={"X-Auth-Token": session_x_auth_token}, verify=False)
+
+
+    def dell_redfish_get_network_interfaces(self, server_name_rib, server_name_short, session_x_auth_token):
+        nic_list = list()
+        url = f"https://{server_name_rib}/redfish/v1/Systems/System.Embedded.1/EthernetInterfaces/"
+        myrequest = requests.get(url, headers={"X-Auth-Token": session_x_auth_token}, verify=False)
+        myjsondata = myrequest.json()
+        myjson_members_data = myjsondata["Members"]
+        myrequest.close()
+        for item in range(0, len(myjson_members_data)):
+            nic = myjsondata["Members"][item]["@odata.id"]
+            nic_url = f"https://{server_name_rib}{nic}"
+            nic_request = requests.get(nic_url, headers={"X-Auth-Token": session_x_auth_token}, verify=False)
+            nicdata = nic_request.json()
+            if nicdata["MACAddress"] != "":
+                nic_list.append(nicdata["Description"])
+                nic_list.append(self.nic_port_mapping(nicdata["Description"]))
+                if nicdata["MACAddress"] != "":
+                    nic_list.append(nicdata["MACAddress"])
+                
+            nic_request.close()
+        self.netbox_server_dict[server_name_short].update({"nics": nic_list})
+
+
+    def server_redfish_get_system_info(self, server_name_idrac, server_name_short, session_x_auth_token, hw_vendor):
+        url = f"https://{server_name_idrac}{self.redfish_get_info_mapping.get(hw_vendor)}"
+        myrequest = requests.get(url, headers={"X-Auth-Token": session_x_auth_token}, verify=False)
+        myjsondata = myrequest.json()
+        myrequest.close()
+        # print(f"\tSimple Memory check ...{myjsondata["MemorySummary"]["Status"]["Health"]}")
+        memory = myjsondata["MemorySummary"]["TotalSystemMemoryGiB"]
```

Comment:
While that does exist it tells us on a dell:
>  "Status@Redfish.Deprecated": "Please migrate to use Status in the individual Memory resources"
But iterating over all sticks might be a bit cumbersome. Probably okay for now

Reply:
```suggestion
        memory_gib = myjsondata["MemorySummary"]["TotalSystemMemoryGiB"]
```
They tell us the unit, we should use it.